add RMSE and MAE metrics.

dnaveenr opened the pull request:
This PR adds RMSE - Root Mean Squared Error and MAE - Mean Absolute Error to the metrics API.
Both implementations are based on usage of sciket-learn.

Feature request here : Add support for continuous metrics (RMSE, MAE) [#3608](https://github.com/huggingface/datasets/issues/3608)

Please suggest any changes if required. Thank you.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/datasets/pr_3845). All of your documentation changes will be reflected on that endpoint.

dnaveenr commented:
@mariosasko I've reopened it here. Please suggest any changes if required. Thank you.

mariosasko commented on metrics/rmse/rmse.py:
```diff
@@ -0,0 +1,80 @@
+            ],
+        )
+
+    def _compute(self, predictions, references):
+
+        rmse = mean_squared_error(references, predictions, squared=False)
+
+        return {"rmse": rmse}
```
Let's also add the missing params here (+ docstring in `KWARGS_DESCRIPTION`):
```suggestion
    def _compute(self, predictions, references, sample_weight=None, multioutput="uniform_average", squared=True):

        mse = mean_squared_error(references, predictions, sample_weight=sample_weight, multioutput=multioutput, squared=squared)
        
        return {"mse": mse}
```

mariosasko reviewed (changes requested):
Great job so far!

dnaveenr commented:
Thanks for suggestions. :) I have added update the KWARGS_DESCRIPTION for the missing params and also changed RMSE to MSE.
While testing, I noticed that when the input is a list of lists, we get an error :
`TypeError: float() argument must be a string or a number, not 'list'`
Could you suggest the datasets.Value() attribute to support both list of floats and list of lists containing floats ?

mariosasko commented:
Just add a new config to cover that case. You can do this by replacing the current `features` dict with:
```python
features=datasets.Features(
    {
        "predictions": datasets.Sequence(datasets.Value("float")),
        "references": datasets.Sequence(datasets.Value("float")),
    }
    if self.config_name == "multioutput"
    else {
        "predictions": datasets.Value("float"),
        "references": datasets.Value("float"),
     }
),
```
Feel free to suggest a better name for the config than `multioutput`

mariosasko commented:
Also, could you please move the changes to a new branch and open a PR from there (for the 3rd time 😄) because the diff shows changes from unrelated PRs (maybe due to rebasing?).

dnaveenr commented:
Thanks for the input, I have added new config to support for multi-dimensional lists and updated the examples as well.

Sure. Will do that and open a new PR for these changes.
